7903374: Feature Tests - Adding six JavaTest GUI legacy automated feature test scripts

gollayadav opened the pull request:
Adding below automated legacy JavaTest GUI feature Test Scripts to the Jemmy regression suite and tested locally on three platforms(Linux, Windows, Mac OS) and working fine. Also verified with JDK8 on macos.

1.CustomReport6.java
2.CustomReport7.java
3.ReportCreate2.java
4.ReportCreate3.java
5.Markers2.java
6.Markers3.java

dbessono reviewed (changes requested):
Please correct formatting (e.g. Markers.java) and add newlines at the end of all files

gollayadav commented:
> Please correct formatting (e.g. Markers.java) and add newlines at the end of all files

Thank you for reviewing. Corrected the format and added new lines at the end of all files. Could you please check now

dbessono commented on gui-tests/src/gui/src/jthtest/Markers/Markers.java:
```diff
@@ -0,0 +1,463 @@
+import org.netbeans.jemmy.operators.JMenuOperator;
+import org.netbeans.jemmy.operators.JPopupMenuOperator;
+
+public class Markers extends ConfigTools {
```
Please add javadoc for this class containing the explanation of the purpose.

dbessono commented on gui-tests/src/gui/src/jthtest/Markers/Markers.java:
```diff
@@ -0,0 +1,463 @@
+import org.netbeans.jemmy.operators.JMenuOperator;
+import org.netbeans.jemmy.operators.JPopupMenuOperator;
+
+public class Markers extends ConfigTools {
```
This class seem to contain a collection of static methods. Why is it inherited?

sycob replied:
@gollayadav cannot see the answer to the quesiton or any explaination or updates

dbessono replied:
@gollayadav cannot see the answer to the quesiton or any explaination or updates

gollayadav replied:
> @gollayadav cannot see the answer to the quesiton or any explaination or updates

Sorry for missing this question!!!

'Markers.java' inherited 'ConfigTools.java' because it has the generic methods(For example: 'openLoadConfigDialogByMenu' - This method is used to load the configuration(.jti) file) related to javatest configuration editor and these methods are called directly from the Markers test scripts because the 'Markers' functionality/option depends on configuration editor menu.

Also, these generic methods in ConfigTools.java are used in other test scripts as well (Example: CreateWorkdir) and invoked by using the class name.

dbessono replied:
The question is specifically about having these methods of class Markers _static_. If they are kept static then static imports could be used instead inheritance. Or if they are supposed to be used as inherited methods then they should be made non-static.

gollayadav replied:
Thanks again for your review comments.

Yes, this can be done and I have used the static imports in the test scripts. Changes are available now and the build is also successful. Could you please check it once?

dbessono replied:
Checked the test run again. Successful. Thanks for using static imports in two contributed classes. Looks good.

dbessono commented:
> > Please correct formatting (e.g. Markers.java) and add newlines at the end of all files
> 
> Thank you for reviewing. Corrected the format and added new lines at the end of all files as suggested. Could you please check now

Source formatting is still not correct. Please move the javadocs for the test classes to the appropriate location.

dbessono commented on gui-tests/src/gui/src/jthtest/CustomReport/CustomReport7.java:
```diff
@@ -0,0 +1,40 @@
+ * This test case verifies that report can be generated in html,xml and plain
+ * text format.
+ */
+
```
This line should be also better removed. The same applies to CustomReport6.

dbessono commented on gui-tests/src/gui/src/jthtest/Markers/Markers3.java:
```diff
@@ -0,0 +1,61 @@
+ */
+package jthtest.Markers;
+
+/**
+ * This test case verifies that a marked question has a small icon to indicate that it has been marked.
+ */
```
Please place this javadoc for the class correctly. This applied to other classes.

gollayadav commented:
@dbessono Thanks again for your time in reviewing.
Modified/Updated the test scripts as suggested above. Could you please check once now.

gollayadav commented:
/integrate

dbessono commented:
/sponsor
